api/v1alpha1/gateway_types.go

```diff
@@ -158,20 +180,41 @@ const (
 // - gcp: https://cloud.google.com/load-balancing/docs/use-ssl-policies#creating_an_ssl_policy_with_a_custom_profile
 // - aws: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html#describe-ssl-policies
 // - azure: https://docs.microsoft.com/en-us/azure/app-service/configure-ssl-bindings#enforce-tls-1112
-type ListenerTLS struct {
+type TLSConfig struct {
 	// Certificates is a reference to one or more Kubernetes objects each containing
-	// an identity certificate that is bound to the listener. The hostname in a TLS
+	// an identity certificate that is bound to a listener. The hostname in a TLS
 	// SNI client hello message is used for certificate matching and route hostname
-	// selection. The SNI server_name must match a route hostname for the Gateway to
-	// route the TLS request.
+	// selection.
 	//
 	// If apiGroup and kind are empty, will default to Kubernetes Secrets resources.
 	//
 	// Support: Core (Kubernetes Secrets)
 	// Support: Implementation-specific (Other resource types)
 	//
-	// +required
-	Certificates []core.TypedLocalObjectReference `json:"certificates"`
+	// +optional
+	ListenerCertificates []core.TypedLocalObjectReference `json:"listenerCertificates,omitempty"`
+	// CACertificates is a reference to one or more Kubernetes objects
+	// each containing a CA certificate used by the TLS client for
+	// establishing a connection with a server.
+	//
+	// Here is a ConfigMap example (in yaml):
+	//
+	// apiVersion: v1
+	// kind: ConfigMap
+	// metadata:
+	//  name: my-dest-svc-ca
+	//  namespace: my-dest-svc-namespace
+	//  data:
+	//    ca-bundle.crt: |
+	//      -----BEGIN CERTIFICATE-----
+	//      Destination Service CA Certificate Bundle.
+	//      -----END CERTIFICATE-----
+	//
+	// Support: Core (Kubernetes ConfigMap)
+	// Support: Implementation-specific (For other resource types)
```

Comment:
`s/Implementation-specific/Custom` to be consistent with how this is used in other places.
Just a note: https://github.com/kubernetes-sigs/service-apis/issues/56